lib/discovery_api/search/storage.ex

```diff
@@ -0,0 +1,73 @@
+defmodule DiscoveryApi.Search.Storage do
+  @moduledoc """
+  Optimize searches by looking up searched words in a table where the searched word is the key and the value is the dataset.
+  """
+  use GenServer
+
+  alias DiscoveryApi.Data.Model
+
+  @all_punctuation ~r/[^\w\s]/
+
+  @spec index(%DiscoveryApi.Data.Model{}) :: :ok
+  def index(model) do
+    GenServer.cast(__MODULE__, {:index, model})
+  end
+
+  @spec search(String.t()) :: list(%DiscoveryApi.Data.Model{})
+  def search(query) do
+    query
+    |> String.split()
+    |> Enum.map(&remove_punctuation/1)
+    |> Enum.map(&lookup_word/1)
+    |> Enum.map(&MapSet.new/1)
```

Comment:
These make sense to me, but it might be more readable for someone like Nathaniel or Austin if this were a single function with a descriptive name of what it's doing.

Reply:
```suggestion
    |> Enum.map(&get_datasets_for_word/1)
    |> Enum.map(&and_datasets_together/1)
```

I'm fine with just the first line and leaving the MapSet - like I said, I get it, but I'm a princi"paL"

Reply:
When it comes to this pipeline, I'm not a princiPAL; I'm indifferent.